Core/ContentBlockerRulesManager.swift

```diff
@@ -23,43 +23,75 @@ import os.log
 import TrackerRadarKit
 
 public class ContentBlockerRulesManager {
-    
+
+    private static let rulesIdentifier = "tds"
+
     public static let shared = ContentBlockerRulesManager()
     
     public var blockingRules: WKContentRuleList?
 
     private init() {}
 
-    public func compileRules(completion: ((WKContentRuleList?) -> Void)?) {
-        guard let trackerData = TrackerDataManager.shared.trackerData else {
-            completion?(nil)
-            return
+    public func recompile() {
+        guard let store = WKContentRuleListStore.default() else {
+            fatalError("Failed to access the default WKContentRuleListStore")
         }
 
-        let storageCache = StorageCacheProvider().current
-        let unprotectedSites = UnprotectedSitesManager().domains
-        let tempUnprotectedDomains = storageCache.fileStore.loadAsArray(forConfiguration: .temporaryUnprotectedSites)
-            .filter { !$0.trimWhitespace().isEmpty }
+        DispatchQueue.global(qos: .background).async {
+            store.removeContentRuleList(forIdentifier: "tds") { _ in
```

Comment:
The constant could be used here instead:

```suggestion
            store.removeContentRuleList(forIdentifier: Self.rulesIdentifier) { _ in
```